.changeset/cool-foxes-talk.md

```diff
@@ -0,0 +1,34 @@
+---
+"astro": minor
+---
+
+Adds new helper functions for integration developers.
+
+- Symbols used to represent `Astro.locals` and `Astro.clientAddress` are now available as static properties on the `App` class: `App.Symbol.locals` and `App.Symbol.clientAddress`.
+
+- `Astro.clientAddress` can now be passed directly to the `app.render()` method.
+```ts
+const response = await app.render(request, { clientAddress: "012.123.23.3" })
+```
+
+- Helper functions for converting node http request and response objects to web-compatible `Request` and `Response` objects are now provided as static methods on the `NodeApp` class.
```

Comment:
```suggestion
- Helper functions for converting Node.js HTTP request and response objects to web-compatible `Request` and `Response` objects are now provided as static methods on the `NodeApp` class.
```

> HTTP request and response objects

I suggest being more explicit. I suppose you meant `http.IncomingMessage`? Or did you mean something else?

Reply:
You're right, I meant to refer to the classes from `node:http`.